src/riak_cs_block_server.erl

```diff
@@ -307,24 +309,24 @@ handle_local_notfound(RcPid, FullBucket, FullKey, GetOptions2,
         {error, Why} when Why == disconnected;
                           Why == timeout ->
             _ = lager:debug("get_block_local/5 failed: {error, ~p}", [Why]),
-            RetryFun(NumRetries + 1);
+            RetryFun([{local_quorum, Why}|ErrorReasons]);
 
         {error, notfound} when UseProxyGet andalso ProxyActive->
             case get_block_remote(RcPid, FullBucket, FullKey,
                                   ClusterID, GetOptions2) of
                 {ok, _} = Success ->
                     ProceedFun(Success);
-                {error, _} ->
-                    RetryFun(NumRetries + 1)
+                {error, Reason} ->
+                    RetryFun([{remote_quorum, Reason}|ErrorReasons])
             end;
         {error, notfound} when UseProxyGet ->
-            RetryFun(NumRetries + 1);
+            RetryFun([{local_quorum, notfound}|ErrorReasons]);
         {error, notfound} ->
-            RetryFun(failure);
+            RetryFun([{local_quorum, notfound}|ErrorReasons]);
```

Comment:
This branch indicates 
- get object failed by notfound with default N (usually 3) and
- proxy get is disabled or not-used (because local cluster is origin).

There is no strong reason that retry make the operation succeed after this error.
Original "immediately fail" seems reasonable.